src/Map.jsx

```diff
@@ -109,33 +111,40 @@ class Map extends Component {
    */
   getCoordinateArray(stop) {
     console.log(this);
-    return [stop.lat, stop.lon];
+    return [stop.lon, stop.lat];
   }
   /**
    * given the two selected stop sids, returns a line segment
    * between them
    */
-  getRouteBetweenStops(routeStops) {
-    const stopSids = this.state.selectedStops.map(stop => stop.sid);
+  getRouteBetweenStops(routeStops, stops) {
+    const stopSids = stops.map(stop => stop.sid);
     stopSids.sort((a, b) => a - b);
     const route = routeStops.map(stop => this.getCoordinateArray(stop));
     let startingPoint = routeStops.find(stop => stop.sid === stopSids[0]);
+    let endingPoint = routeStops.find(stop => stop.sid === stopSids[1]);
+    /*
+    * if either value is unedfined, it means user selected another stop on another route.
+    * so clear all stops and subroute
+    */
+    if (typeof startingPoint === 'undefined' || typeof endingPoint === 'undefined') {
```

Comment:
@EddyIonescu this handles the previous error of selecting stops from multiple routes

Reply:
Ok, thanks for leaving a comment to clarify this.